src/components/MainNav/MainNav.js

```diff
@@ -27,198 +22,93 @@ import { CurrentAppGroup } from './CurrentApp';
+        const queryState = getQueryResourceState(module, stripes.store);
 
         // only update location if query state has changed
         if (!isEqual(queryState, prevQueryState[moduleName])) {
-          curQuery = updateLocation(module, curQuery, this.store, history, location);
+          curQuery = updateLocation(module, curQuery, stripes.store, history, location);
           prevQueryState[moduleName] = queryState;
         }
       }
     });
 
     // remove QueryProvider cache to be 100% sure we're starting from a clean slate.
-    this.props.queryClient.removeQueries();
-  }
-
-  componentDidUpdate(prevProps) {
-    const { modules, location } = this.props;
-    this.curModule = getCurrentModule(modules, location);
-    if (this.curModule && !isEqual(location, prevProps.location)) {
-      updateQueryResource(location, this.curModule, this.store);
-    }
-  }
-
-  componentWillUnmount() {
-    this._unsubscribe();
-  }
-
-  toggleUserMenu() {
-    const isOpen = this.state.userMenuOpen;
-    this.setState({
-      userMenuOpen: !isOpen,
-    });
-  }
-
-  getAppList(lastVisited) {
-    const { stripes, location: { pathname }, modules, intl: { formatMessage } } = this.props;
-
-    const apps = modules.app.map((entry) => {
-      const name = entry.module.replace(packageName.PACKAGE_SCOPE_REGEX, '');
-      const perm = `module.${name}.enabled`;
+    queryClient.removeQueries();
 
-      if (!stripes.hasPerm(perm)) {
-        return null;
-      }
-
-      const id = `clickable-${name}-module`;
-
-      const pathRoot = pathname.split('/')[1];
-      const entryRoot = entry.route.split('/')[1];
-      const active = pathRoot === entryRoot;
-
-      const last = lastVisited[name];
-      const home = entry.home || entry.route;
-      const href = (active || !last) ? home : lastVisited[name];
-
-      return {
-        id,
-        href,
-        active,
-        name,
-        ...entry,
-      };
-    }).filter(app => app);
-
-    /**
-     * Add Settings to apps array manually
-     * until Settings becomes a standalone app
-     */
-
-    if (stripes.hasPerm('settings.enabled')) {
-      apps.push({
-        displayName: formatMessage({ id: 'stripes-core.settings' }),
-        id: 'clickable-settings',
-        href: lastVisited.x_settings || '/settings',
-        active: pathname.startsWith('/settings'),
-        description: 'FOLIO settings',
-        iconData: {
-          src: settingsIcon,
-          alt: 'Tenant Settings',
-          title: 'Settings',
-        },
-        route: '/settings'
-      });
+    return () => {
+      _unsubscribe();
+    };
+  }, []); // eslint-disable-line
+
+  useEffect(() => {
+    setSelectedApp(apps.find(entry => entry.active));
+    const nextCurModule = getCurrentModule(modules, location);
+    if (nextCurModule) {
+      setCurModule(getCurrentModule(modules, location));
+      updateQueryResource(location, nextCurModule, stripes.store);
     }
```

Comment:
True confession: I'm lost here. nextCur, setCur, getCurrent ... they're all very similar. Maybe header-comments to explain what each `useEffect()` block does?